diesel/Cargo.toml

```diff
@@ -55,12 +55,14 @@ huge-tables = ["64-column-tables"]
 postgres = ["pq-sys", "postgres_backend"]
 sqlite = ["libsqlite3-sys", "diesel_derives/sqlite"]
 mysql = ["mysqlclient-sys", "url", "percent-encoding", "bitflags", "mysql_backend"]
-without-deprecated = []
-with-deprecated = []
+without-deprecated = ["diesel_derives/without-deprecated"]
+with-deprecated = ["diesel_derives/with-deprecated"]
 network-address = ["ipnetwork", "libc"]
 numeric = ["num-bigint", "bigdecimal", "num-traits", "num-integer"]
 postgres_backend = ["diesel_derives/postgres", "bitflags", "byteorder"]
 mysql_backend = ["diesel_derives/mysql", "byteorder"]
+i-implement-a-third-party-backend-and-opt-into-breaking-changes = []
+doc_cfg = []
```

Comment:
I would personally use cfgs rather than features for this:
- `doc_cfg` is not meant for anyone actually using the crate
- It provide a slight speed bump to people using the breaking changes feature

Reply:
Yeah I would also recommend a `--cfg` here. You can check out Tokio's `Cargo.toml` to see how to enable it on docs.rs.